lib/mirage.ml

```diff
@@ -33,6 +33,83 @@ let driver_error name =
   Printf.sprintf "fail (Failure %S)" name
 
 (** {2 Devices} *)
+type qrexec = job
+let qrexec = job
+
+let qrexec_qubes = impl @@ object
+  inherit base_configurable
+  method ty = qrexec
+  val name = Name.ocamlify @@ "qrexec_"
+  method name = name
+  method module_name = "Qubes.RExec"
+  method packages = Key.pure ["mirage-qubes"]
+  method libraries = Key.pure ["mirage-qubes"]
+  method connect i modname _args =
+  match Key.(get (Info.context i) target) with
+  | `Unix | `MacOSX | `Xen ->
+  Fmt.strf
+  "return (`Ok ())"
```

Comment:
I agree, the best is to use the `configure` method for that purpose (until we have a better solution, see https://github.com/mirage/functoria/issues/12). 

In any cases, having the two connect implementation that returns different types is a bad idea.